rcl_yaml_param_parser/src/parser.c

```diff
@@ -1278,6 +1659,201 @@ static rcutils_ret_t parse_value(
   return ret;
 }
 
+static rcutils_ret_t parse_descriptor(
+  namespace_tracker_t * ns_tracker,
+  const yaml_event_t event,
+  const bool is_seq,
+  const size_t node_idx,
+  const size_t parameter_idx,
+  rcl_params_t * params_st)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(params_st, RCUTILS_RET_INVALID_ARGUMENT);
+
+  rcutils_allocator_t allocator = params_st->allocator;
+  RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
+    &allocator, "invalid allocator", return RCUTILS_RET_INVALID_ARGUMENT);
+
+  if (0U == params_st->num_nodes) {
+    RCUTILS_SET_ERROR_MSG("No node to update");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+
+  const size_t val_size = event.data.scalar.length;
+  const char * value = (char *)event.data.scalar.value;
+  yaml_scalar_style_t style = event.data.scalar.style;
+  const uint32_t line_num = ((uint32_t)(event.start_mark.line) + 1U);
+
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    value, "event argument has no value", return RCUTILS_RET_INVALID_ARGUMENT);
+
+  if (style != YAML_SINGLE_QUOTED_SCALAR_STYLE &&
+    style != YAML_DOUBLE_QUOTED_SCALAR_STYLE &&
+    0U == val_size)
+  {
+    RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING("No value at line %d", line_num);
+    return RCUTILS_RET_ERROR;
+  }
+
+  if (NULL == params_st->descriptors[node_idx].parameter_descriptors) {
+    RCUTILS_SET_ERROR_MSG("Internal error: Invalid mem");
+    return RCUTILS_RET_BAD_ALLOC;
+  }
+
+  rcl_param_descriptor_t * param_descriptor =
+    &(params_st->descriptors[node_idx].parameter_descriptors[parameter_idx]);
+
+  data_types_t val_type;
+  void * ret_val = get_value(value, style, &val_type, allocator);
+  if (NULL == ret_val) {
+    RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      "Error parsing value %s at line %d", value, line_num);
+    return RCUTILS_RET_ERROR;
+  }
+
+  if (NULL == ns_tracker->parameter_ns) {
+    RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      "Parameter assignment at line %d unallowed in parameter__descriptors", line_num);
```

Comment:
@bpwilcox here and elsewhere, isn't it the point of having a `PARAMS_DESCRIPTORS_KEY` macro to not use the string literal everywhere?

Reply:
You're referring to why I am using the `ns_tracker->descriptor_key_ns` in this function?

Reply:
> You're referring to why I am using the `ns_tracker->descriptor_key_ns` in this function?

Not exactly, that I understand. What I meant to say is, let's try use `PARAMS_DESCRIPTORS_KEY` instead of `parameter_descriptors` where possible. For this particular line:

```c
RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
      "Parameter assignment at line %d unallowed in " PARAMS_DESCRIPTORS_KEY, line_num);
```

Sorry for not being clear enough.

Reply:
I understand now, thanks for clarifying